src/main/java/uk/gov/hmcts/divorce/solicitor/event/Applicant1SolicitorViewApplicant2ContactDetails.java

```diff
@@ -58,15 +57,18 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDeta
             APPLICANT_1_SOLICITOR_VIEW_APPLICANT_2_CONTACT_INFO, details.getId()
         );
 
-        List<String> errors = new ArrayList<>();
-
         boolean applicantIsConfidential = details.getData().getApplicant2().isConfidentialContactDetails();
         if (applicantIsConfidential) {
-            errors.add(CONFIDENTIAL_APPLICANT_ERROR);
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .errors(Collections.singletonList(CONFIDENTIAL_APPLICANT_ERROR))
+                .build();
         }
 
+        var applicant2 = details.getData().getApplicant2();
+        applicant2.setNonConfidentialAddress(applicant2.getAddress());
+
         return AboutToStartOrSubmitResponse.<CaseData, State>builder()
-            .errors(errors)
+            .data(details.getData())
```

Comment:
did the errors not propagate as they were?

Reply:
`.errors(errors)` returns the error text to the user. I changed the code to return the error response immediately on line 63 if validation fails, and no longer have validation errors to include in the response here